### ISSUE TITLE
bug(python): Arrow table from polars can't be read and queried the same as pandas

### ISSUE THREAD

**Issue by prrao87** [lancedb/lancedb#791]
### LanceDB version

0.4.3

### What happened?

The following code snippets from pandas and polars should work the same way, but the polars snippet fails (the only difference is that in Polars, we currently have to transform to an arrow table and then to a LanceDB table, so we have to specify the schema).

### Pandas

```python
import lancedb
import pandas as pd
import shutil

db = lancedb.connect("./test_lancedb")

data = pd.DataFrame({
    "vector": [[3.1, 4.1], [5.9, 26.5]],
    "item": ["foo", "bar"],
    "price": [10.0, 20.0]
})

try:
    shutil.rmtree("./test_lancedb")
    tbl = db.create_table("test_table", data=data)
except OSError:
    tbl = db.open_table("test_table")

query_vector = [5.9, 26.5]
print(tbl.search(query_vector).to_pandas())

"""
        vector item  price   _distance
0  [5.9, 26.5]  bar   20.0    0.000000
1   [3.1, 4.1]  foo   10.0  509.599976
"""
```

### Polars

```python
import lancedb
import polars as pl
import shutil

db = lancedb.connect("./test_lancedb")

data = pl.DataFrame({
    "vector": [[3.1, 4.1], [5.9, 26.5]],
    "item": ["foo", "bar"],
    "price": [10.0, 20.0]
})

arrow_data = data.to_arrow()

try:
    shutil.rmtree("./test_lancedb")
    tbl = db.create_table("test_table", data=data, schema=arrow_data.schema)
except OSError:
    tbl = db.open_table("test_table")

query_vector = [5.9, 26.5]
print(tbl.search(query_vector).to_list())
```

#### Error

```
Traceback (most recent call last):
  File "/Users/prrao/code/lancedb/docs/test.py", line 22, in <module>
    print(tbl.search(query_vector).to_list())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/prrao/code/lancedb/venv/lib/python3.11/site-packages/lancedb/query.py", line 243, in to_list
    return self.to_arrow().to_pylist()
           ^^^^^^^^^^^^^^^
  File "/Users/prrao/code/lancedb/venv/lib/python3.11/site-packages/lancedb/query.py", line 437, in to_arrow
    return self._table._execute_query(query)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/prrao/code/lancedb/venv/lib/python3.11/site-packages/lancedb/table.py", line 1123, in _execute_query
    return ds.to_table(
           ^^^^^^^^^^^^
  File "/Users/prrao/code/lancedb/venv/lib/python3.11/site-packages/lance/dataset.py", line 325, in to_table
    return self.scanner(
           ^^^^^^^^^^^^^
  File "/Users/prrao/code/lancedb/venv/lib/python3.11/site-packages/lance/dataset.py", line 253, in scanner
    builder = builder.nearest(**nearest)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/prrao/code/lancedb/venv/lib/python3.11/site-packages/lance/dataset.py", line 1603, in nearest
    raise TypeError(
TypeError: Query column vector must be a vector. Got large_list<item: double>.
```

I think what's happening is that when we convert the polars dataframe to a pyarrow schema, the type of the input gets transformed to a `large_list` in Rust, which is then not correctly recognized in LanceDB when we try to submit a Python query vector object (the query fails regardless of whether we provide a Python list or a numpy array). 

### Desired behaviour

Ideally (until native Polars support is available), the same workflow in which we provide input vector as Python lists/numpy arrays that works for pandas should also work for Polars, despite using PyArrow as the intermediary.

**Comment by westonpace** [comment]
It looks like polars doesn't have a concept of "fixed size list" either.  However, we are already coercing all list types to be fixed size list when coming from arrow so I think it would be ok to coerce "large_list" to fixed size list too.

**Comment by prrao87** [comment]
@westonpace I thought Ritchie settled on the `Array` data type to supercede `FixedSizeList` in Polars and [that was merged](https://github.com/pola-rs/polars/pull/8943) several months ago? I think the problem here is that we don't yet have native Polars integration (which should support the `Array` type in Polars), and till then, the `large_list` is inferred from the PyArrow schema because we use that as an intermediary.

**Comment by westonpace** [comment]
Ah, you are probably right, I don't follow polars super closely.  It does look like the `Array` type works:

```
data = pl.DataFrame({
    "vector": pl.Series("vector", [[3.1, 4.1], [5.9, 26.5]], dtype=pl.Array(pl.Float32, 2)),
    "item": ["foo", "bar"],
    "price": [10.0, 20.0]
})
```

But if we want compatibility we should still coerce `List` to `FixedSizeList` in the same way we do with pandas.

**Comment by prrao87** [comment]
```py
data = pl.DataFrame({
    "vector": pl.Series("vector", [[3.1, 4.1], [5.9, 26.5]], dtype=pl.Array(pl.Float32, 2)),
    "item": ["foo", "bar"],
    "price": [10.0, 20.0]
})
```

This solution helps construct the DataFrame, but doesn't help at query time because we'd then have to do something really ugly like `pl.Series("vector", [[2.1, 3.1]).to_numpy()` so that LanceDB can then use it. Lots of unnecessary data transformations on the user's side.

**Comment by changhiskhan** [comment]
i think we have some explicit sanitization code that should convert variable list to fixed size list if the column is named `vector`. I need to look at the source code but i'm hoping it's as simple as adding another condition for large list

**Comment by changhiskhan** [comment]
ok yeah, if you look at https://github.com/lancedb/lancedb/blob/main/python/lancedb/table.py#L1279 i *think* we just need to also check `pa.types.is_large_list`

**Comment by prrao87** [comment]
Great! If that's fixed then we should be able to just pass in a Python list/numpy array and LanceDB's arrow conversion kicks in to handle the type inconsistencies? As of now, none of my Polars snippets seem to be working 😅

**Comment by prrao87** [comment]
This is fixed as per #811. 🚀
